src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Rendering/EventForDotNet.ts

```diff
@@ -69,18 +74,137 @@
       case 'pointerout':
       case 'pointerover':
       case 'pointerup':
-        return new EventForDotNet<UIPointerEventArgs>('pointer', { Type: event.type });
+        return new EventForDotNet<UIPointerEventArgs>('pointer', parsePointerEvent(event));
 
+      case 'wheel':
       case 'mousewheel':
-        return new EventForDotNet<UIWheelEventArgs>('wheel', { Type: event.type });
-
+        return new EventForDotNet<UIWheelEventArgs>('wheel', parseWheelEvent(event));
 
       default:
-        return new EventForDotNet<UIEventArgs>('unknown', { Type: event.type });
+        return new EventForDotNet<UIEventArgs>('unknown', { type: event.type });
     }
   }
 }
 
+function parseDragEvent(event: any) {
+  return {
+    type: event.type,
+    detail: event.detail,
+    dataTransfer: event.dataTransfer,
+    screenX: event.screenX,
+    screenY: event.screenY,
+    clientX: event.clientX,
+    clientY: event.clientY,
+    button: event.button,
+    buttons: event.buttons,
+    mozPressure: event.mozPressure,
+    ctrlKey: event.ctrlKey,
+    shiftKey: event.shiftKey,
+    altKey: event.altKey,
+    metaKey: event.metaKey
+  }
+}
+
+function parseWheelEvent(event: any) {
+  return Object.assign(parseMouseEvent(event),
+    {
+      deltaX: event.deltaX,
+      deltaY: event.deltaY,
+      deltaZ: event.deltaZ,
+      deltaMode: event.deltaMode
+    });
+}
+
+function parseProgressEvent(event: any) {
+  return {
+    type: event.type,
+    lengthComputable: event.legthComputable,
```

Comment:
Typo in `legthComputable`. 

This indicates that we should be using TypeScript types to avoid the risk of errors like that. In this example, `event` could be typed as `ProgressEvent`. Would it be possible to strongly-type all the `event` objects we're using in this file?

Reply:
Agreed. Will do that.

Reply:
Just to make sure we're talking about the same thing, TypeScript has *built-in* types for all these DOM events (e.g., `ProgressEvent`, `PointerEvent`, etc. are built into the standard library). You don't need to write those type definitions yourself.

Reply:
(However I do agree with your decision to map a specific subset of properties out of them into the `UiXyzEventArgs` types)

Reply:
Yeah, I know. That is the reason I made the types myself. Some of them will not be useful in Blazor _today_ like the ones which are references to other objects. 

We can ofc in the future drop those as they are there just to guide ourselves and no public API would be affected.

Reply:
Btw, I only left the `parseDragEvent(event: any)` as `any` because the `datatransfer` object is really tricky so to make it easier I left it `any`. However, I did the test before changing to `DragEvent` and no types were found :)